### PR TITLE
Add 96boards Nitrogen, board id 0x9014

### PR DESCRIPTION
**Pull request by rsalveti**
Board details available at https://www.96boards.org/product/nitrogen

Signed-off-by: Ricardo Salveti <ricardo@opensourcefoundries.com>

### CODE REVIEW

File: records/board/96b_nitrogen.yaml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+common:
+    macros:
+        - IO_CONFIG_OVERRIDE
```
Comment (c1728p9) [inline records/board/96b_nitrogen.yaml]: What IO customization did you need to make for this board? Please comment in IO_Config_Override.h which pins changed from the default.

File: test/info.py [hunk 2]
```diff
@@ -140,6 +140,7 @@
     (   0x9011,     'lpc11u35_archmax_if',                      None,               'Seeed-Arch-Max'                        ),
     (   0x9012,     'lpc11u35_tiny_if',                         None,               'Seeed-Tiny-BLE'                        ),
     (   0x9013,     'lpc11u35_archlink_if',                     None,               'Seeed-Arch-Link'                       ),
+    (   0x9014,     'lpc11u35_96b_nitrogen_if',                 None,               'Seeed-96Boards-Nitrogen'               ),
```
Comment (c1728p9) [inline test/info.py]: If you want this as part of the next DAPLink release you'll need to add it to the above PROJECT_RELEASE_INFO table as well.

Comment (rsalveti) [reply test/info.py]: Added to the PROJECT_RELEASE_INFO table as suggested, thanks!

File: records/board/96b_nitrogen.yaml (hunk 1 above)
Comment (rsalveti) [reply records/board/96b_nitrogen.yaml]: Added "Nitrogen override" for the pins that changed from the default.

**Comment by c1728p9** [comment]
Thanks for the updates @rsalveti

**Pull request merged**